### PR TITLE
Feature/home order status

### PR DESCRIPTION
**Pull request by maharabhossain1**
Changes:
  - Implemented order status statistics on the home page.
  - Created a new constants file for Four order status.
  - Developed a utility function to adapt backend data for our UI.
  - Introduced UI reusable components: BOX, SkeletonLoader, CircleIcon.
  - Updated the icons folder to include new icons.
  - Utilized Axios for API calls.
  - Installed the Axios package and configured it in apiService.js.
  - Dynamically fetched and displayed data from the API in the UI.

Purpose:
Added order status statistics to the home page, organized constants, created utility functions for data modification and introduced reusable UI components. Updated the icons folder and integrated Axios for efficient API calls, resulting in dynamic data display.

### CODE REVIEW

File: components/box.jsx [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import { cn } from '@/lib/utils';
+import React, { forwardRef } from 'react';
+
+const Box = forwardRef(({ children, className, style, ...rest }, ref) => {
```
Comment (ShehabAhmedSayem) [inline components/box.jsx:4]: I think the name of the component should be `Card`

File: pages/index.jsx [hunk 2]
```diff
@@ -1,9 +1,57 @@
+import { getOrderStatistics } from '@/apiService';
 import Layout from '@/components/layout';
+import OrderStatusCard from '@/components/order-status-card';
+import OrderStatusCardSkeleton from '@/components/order-status-card-skeleton';
+import { transformOrderData } from '@/lib/utils';
+import { useEffect, useState } from 'react';
 
 export default function Home() {
+  const [orderStat, setOrderState] = useState([]);
```
Comment (ShehabAhmedSayem) [inline pages/index.jsx]: Name and setName is not consistent. Rename as `orderStatistics` , `setOrderStatistics`

File: pages/index.jsx [hunk 3]
```diff
@@ -1,9 +1,57 @@
...
-      <Layout>Portfolio Dashboard for Order Management</Layout>
+      <Layout>
+        <div className="p-4">
+          <h1 className="text-xl">Overview</h1>
+          <div className="grid grid-cols-1 gap-4 mt-4 xl:grid-cols-2">
+            {loading ? (
+              <OrderStatusCardSkeleton />
+            ) : (
+              <section className="grid grid-cols-1 gap-4 sm:grid-cols-2 ">
+                {orderStat.map((orderStatus) => {
```
Comment (ShehabAhmedSayem) [inline pages/index.jsx]: Change it to `{orderStatistics.map((orderStatistic)) = {`

File: components/order-status-card-skeleton.jsx [hunk 4]
```diff
@@ -0,0 +1,22 @@
+import React from 'react';
+import { uuidv4 } from '@/lib/utils';
+import SkeletonLoader from './skeleton-loader';
+
+const OrderStatusCardSkeleton = () => {
```
Comment (ShehabAhmedSayem) [inline components/order-status-card-skeleton.jsx]: Change the name and file name to `OrderStatisticCardSkeleton`

File: components/order-status-card.jsx [hunk 5]
```diff
@@ -0,0 +1,55 @@
+import React from 'react';
+import { ArrowDownFillIcon, ArrowUpFillIcon } from '@/icons';
+import { cn } from '@/lib/utils';
+
+import Box from './box';
+import CircleIcon from './circle-icon';
+
+const OrderStatusCard = ({ title, icon, value, percentage, isPositivePercentage }) => {
```
Comment (ShehabAhmedSayem) [inline components/order-status-card.jsx]: Change the name and file name to `OrderStatisticCard`

File: lib/constants.js [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+  },
+  {
+    title: 'Settings',
+    path: '/settings',
+    icon: <SettingsIcon className="text-2xl" />,
+    disabled: true,
+  },
+];
+
+export const ORDER_STATUS_TYPES = {
```
Comment (ShehabAhmedSayem) [inline lib/constants.js]: Change to `ORDER_STATISTIC_TYPES`

**Review by ShehabAhmedSayem — CHANGES_REQUESTED** [review]

File: components/box.jsx (hunk 1 above)
Comment (maharabhossain1) [reply components/box.jsx:4]: I think the name is correct cause this component works for an alternative as a div. After all, we have a similar style in every div so I make a common Box component that has this common style.

**Pull request merged**